### PR TITLE
make transit work correctly with new feeds.

### PR DESCRIPTION
**Pull request by gknisely**
Costing was messed up for transit.  Updated cost and penalty. They were flipped. sigh.
Added defaults for walking between stops and at the beginning or end of a transit route.  
Tweaks here and there for desired defaults based on real world test cases.

### CODE REVIEW

File: src/sif/dynamiccost.cc [hunk 1]
```diff
@@ -92,12 +92,27 @@ void DynamicCost::SetAllowTransitConnections(const bool allow) {
...
+ * travel between transfers.
+ */
+uint32_t DynamicCost::GetMaxTransferDistanceMM() {
+  return 0;
+}
+
+// This method overrides the weight for this mode.  The higher the value
+// the more the mode is favored.
+float DynamicCost::GetModeWeight() {
+  return 0.0f;
```
Comment (dnesbitt61) [inline src/sif/dynamiccost.cc]: shouldn't this be 1.0f?

Comment (gknisely) [inline src/sif/dynamiccost.cc]: Agree

File: src/sif/transitcost.cc [hunk 2]
```diff
@@ -13,12 +13,13 @@ namespace sif {
...
 
+constexpr float kModeWeight         = 2.0f; // Favor this mode?
 constexpr float kDefaultBusFactor   = 1.0f;
 constexpr float kDefaultBusPenalty  = 0.0f;
 constexpr float kDefaultRailFactor  = 1.0f;
 constexpr float kDefaultRailPenalty = 0.0f;
-constexpr float kDefaultTransferCost = 60.0f;
-constexpr float kDefaultTransferPenalty = 300.0f;  // 5 minute default
+constexpr float kDefaultTransferCost = 300.0f;
+constexpr float kDefaultTransferPenalty = 45.0f;
```
Comment (dnesbitt61) [inline src/sif/transitcost.cc]: are these reversed?

Comment (gknisely) [inline src/sif/transitcost.cc]: no.  closely matches your original hard coded defaults.  Cost transfer_cost = { 300.0f, 60.0f };

**Pull request merged**